### DISCUSSION TITLE
Models for convolution inference

### DISCUSSION THREAD

**Discussion by athowes** [epinowcast/epidist#530] [Ideas]
This issue is drawn from conversation with @zsusswein and @SamuelBrand1 at https://github.com/cdcent/cfa-parameter-estimates/issues/22.

## Topline

* We could look to add new models to estimate delays more indirectly where components of the overall delay are observed then one needs to combine estimates of delays together
* This would leverage existing infrastructure and good practises we have for estimation of delays with censoring and truncation biases so could be a good fit for the package

## Problem background

Let person A be the infector and person B be the infectee in an infector-infectee pair.

* Generation interval (GI): infection of A to infection of B
* Serial interval (SI): time from symptoms of A to symptoms of B
* Incubation period (IP): time from infection to symptoms

Assume that we would like to estimate the GI. However, we do not have direct observations of infection times for person A and person B. Instead we have:

1. Data on time of symptom onset for person A and person B (i.e. data on the SI)
2. Auxiliary data on the IP

Using these two sources of data, one can try to estimate the GI. This supposes that the auxiliary data for the IP can be reasonably applied to estimate the IP of person A and person B. Both sources of data are going to be subject to censoring and truncation biases (like the other data that `epidist` aims to handle). See [Ferretti et al. 2020](https://www.science.org/doi/full/10.1126/science.abb6936#supplementary-materials) for an example of doing this.

<img width="834" alt="image" src="https://github.com/user-attachments/assets/0073c558-49af-4ec1-a8cf-8b5613f3f077">

## Why is could be a good fit for this package

* **Builds on existing strengths of package** We can already handle censoring and truncation well. This is an extension which builds on our core strengths
* **Similar workflow** In Ferretti et al. they fit 3 possible delay distribution families, then use AIC to choose a model. We can support this workflow using what I think are best practise tools for estimation and model comparison (`brms` into `tidybayes` with `loo` and all other Stan infra etc.). We could make it easy for a user to run this analysis, setting the `family` argument, running inference, then using `loo`
* **Support for model structures** We have a set-up where you can fit distributional regression models and specify the model on each distributional parameter. This would allow for quite sophisticated modelling approaches. For example, with this auxilitary data, I can imagine that you might want to fit space-time-demographic specific models such that you can reasonably apply it in the transmission chains you have.

## Other thoughts

* With the Bayesian framework, I think another natural extension that you'd want is to incorperate probabilites of transmission pairs. Even further to link into phylogenetic analysis. This is probably really hard. You could use `brms::make_stancode` then start to build towards this without the constraints of `brms`. I know people are working on this type of thing but don't know the state-of-the-art

## Next steps

* It'll become clearer how this might fit in as the package matures I think. Maybe it could be quite easy at some point if we do a good enough job with the infrastructure
* I think this is probably a good research project, so might be a good fit for someone in an academic role. It could also be a project for a student

**Comment by seabbs** [comment]
@parksw3 I think this is in your wheelhouse and is a good fit

**Comment by seabbs** [comment]
> Models for deconvolution inference

Also is it not convolution

**Comment by athowes** [comment]
> Also is it not convolution

Don't know / mind. Copied from @SamuelBrand1 👼

**Comment by seabbs** [comment]
I feel like we always want to be talking about forward processes hence trying to avoid "deconvolution".

**Comment by SamuelBrand1** [comment]
pdf of $X + Y$ is a convolution... so in my head back inferring to get pdf of $X$ from data on $X + Y$ is a deconvolution.

**Comment by SamuelBrand1** [comment]
From f2f I think @seabbs  was a bit radicalised by some _actual_ deconvolution techniques that used to be used... so I'm happy to go with his naming conventions.

**Comment by parksw3** [comment]
I missed this notification and just found this in my email somehow... let me assign myself and hopefully get to it

**Comment by seabbs** [comment]
Noting that where we think distributions are a mixture we should be able to support this in at least the latent model using the  `mixture()` family

**Comment by athowes** [comment]
Would this be better as a discussion? Doesn't immediately relate to package / would require research before can implement these models.

**Comment by SamuelBrand1** [comment]
+1 to that.
